ENGAGEHF/Onboarding/InvitationCodeError.swift

```diff
@@ -0,0 +1,28 @@
+//
+// This source file is part of the ENGAGE-HF project based on the Stanford Spezi Template Application project
+//
+// SPDX-FileCopyrightText: 2023 Stanford University
+//
+// SPDX-License-Identifier: MIT
+//
+
+import Foundation
+
+
+// Based on: https://github.com/StanfordBDHG/PediatricAppleWatchStudy/pull/54/files
+enum InvitationCodeError: LocalizedError {
+    case invitationCodeInvalid
+    case userNotAuthenticated
+    case generalError(String)
+    
+    var errorDescription: String? {
+        switch self {
+        case .invitationCodeInvalid:
+            NSLocalizedString("The invitation code is invalid or has already been used.", comment: "Invitation Code Invalid")
```

Comment:
Please inspect the latest changes in the PR in the PAWS app to reflect the latest changes if this file in this PR. You can inspect them in the main branch and my PR feedback.